packages/kafka-iam-auth/src/index.ts

```diff
@@ -364,7 +421,26 @@ export const runConsumer = async (
   consumerHandler: (messagePayload: EachMessagePayload) => Promise<void>
 ): Promise<void> => {
   try {
-    await initConsumer(config, topics, consumerHandler);
+    await initCustomConsumer({ config, topics, consumerHandler });
+  } catch (e) {
+    genericLogger.error(
+      `Generic error occurs during consumer initialization: ${e}`
+    );
+    processExit();
+  }
+};
```

Comment:
this would be the refactor with the previous suggestion

```
export const runConsumer = async (
  config: KafkaConsumerConfig,
  topics: string[],
  consumerHandler: (messagePayload: EachMessagePayload) => Promise<void>
): Promise<void> => {
  try {
    const consumerRunConfig = (consumer: Consumer) => ({
      autoCommit: false,
      eachMessage: async (payload: EachMessagePayload) => {
        try {
          await consumerHandler(payload);
          await kafkaCommitMessageOffsets(consumer, payload);
        } catch (e) {
          throw kafkaMessageProcessError(
            payload.topic,
            payload.partition,
            payload.message.offset,
            e
          );
        }
      },
    });
    await initCustomConsumer({ config, topics, consumerRunConfig });
  } catch (e) {
    genericLogger.error(
      `Generic error occurs during consumer initialization: ${e}`
    );
    processExit();
  }
};
```